Zombie disease is easier to spread and deadly in minutes. Zombies heal over time.

tom-leys opened the pull request:
<!-- Please read these guidelines before opening your PR: https://docs.spacestation14.io/en/getting-started/pr-guideline -->
<!-- The text between the arrows are comments - they will not be visible on your PR. -->

## About the PR
After some discussion, this PR is a become a buff to the zombie virus. The infection is far more virulent now.
- The zombie disease is easier to spread, being from 20% (fully clothed) to 80% (nude) per bite. Up from 5 and 50% respectively.
- The zombie disease does quite a lot of DoT and that damage increases over time. You will only survive for about 1 minute before you die.
- Actual death (not just crit) turns you into a zombie. Given time, you will come back to life as zombie after death or critting.

NB: This PR initially hurt zombies in space, which the comments reflect. That has been removed now.

**Media**
<!-- 
PRs which make ingame changes (adding clothing, items, new features, etc) are required to have media attached that showcase the changes.
Small fixes/refactors are exempt.
Any media may be used in SS14 progress reports, with clear credit given.

If you're unsure whether your PR will require media, ask a maintainer.

Check the box below to confirm that you have in fact seen this (put an X in the brackets, like [X]):
-->

Please see the video in the comments below

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**
<!--
Here you can fill out a changelog that will automatically be added to the game when your PR is merged.

Only put changes that are visible and important to the player on the changelog.

Don't consider the entry type suffix (e.g. add) to be "part" of the sentence:
bad: - add: a new tool for engineers
good: - add: added a new tool for engineers

Putting a name after the :cl: symbol will change the name that shows in the changelog (otherwise it takes your GitHub username)
Like so: :cl: PJB
-->

:cl:
- tweak: Zombification happens post death, rather than just crit. So you lie on the floor a bit before beginning your new life.
- tweak: Zombies are more likely (10-40% depending on clothes) to inflict a zombifying Damage over Time.  This was 5% - 50% before. A typical outfit without outer suit, mask or helmet has a 17.5% chance of being infected each hit.
- tweak: Zombies have some heal over time. This might also bring them back from the dead (again).

asperger-sind commented:
I'm fine with zombies taking space damage, but the rest of the changes are imo iffy.
Zombies are already pretty weak provided you know how to deal with them (or just flat out run away, as you're faster)

Zombies instantly discarding helmet/mask prevents stealth zombies (which are to a degree possible as-is) and doesn't really make sense logically; how are you getting that helmet and mask off provided you're a mindless being?

Being stunned for **five** whole seconds upon convert makes chain infections nigh impossible because people'll either kill you in your stun or run off before you can do anything.
I think a better way to solve this would be making the zombifying not instant, so upon crit you sit like that for like 5-10 seconds and then zombify, instead of just getting zombified and THEN stunned.

I'm concerned about this however; if zombies start taking space damage then everyone on-board with any form of EVA compatibility will just go to space and sit there for the rest of the round/until shuttle call.
That won't really spur gameplay nor roleplay,

keronshb commented:
There really is no need for the zombies to lose their helmet or mask. It's potentially possible one could fall off when transforming but the likeliness is very low for every zombie (see: mostly every other zombie game)

Zombies historically don't take damage from pressure in SS13. There are usually ships that are spaced with active zombies on them.

Stunning on conversion is a horrible idea. Especially one for that long.

tom-leys commented:
> Zombies instantly discarding helmet/mask prevents stealth zombies (which are to a degree possible as-is) and doesn't really make sense logically; how are you getting that helmet and mask off provided you're a mindless being?

I'm not sure how zombies bite through masks. I'd prefer you bite them off, but that was more complex.

> 
> Being stunned for **five** whole seconds upon convert makes chain infections nigh impossible 

Agreed. I've reduced this to 2, which seems very short. I just wanted some interval between living and undead. 

> I think a better way to solve this would be making the zombifying not instant, so upon crit you sit like that for like 5-10 seconds and then zombify, instead of just getting zombified and THEN stunned.

I much prefer that, just not quite sure how to implement it. Oh, actually I think I can do that quite easily ...

> I'm concerned about this however; if zombies start taking space damage then everyone on-board with any form of EVA compatibility will just go to space and sit there for the rest of the round/until shuttle call. That won't really spur gameplay nor roleplay,

This is a really fair point. So to balance the DoT of being in space there is a heal over time that applies to zombies globally. So now zombies prefer being inside for the healing and can survive several minutes outside in space.

I've also seriously buffed the infection so that deals DoT to the infected living.

tom-leys commented:
> There really is no need for the zombies to lose their helmet or mask. It's potentially possible one could fall off when  transforming but the likeliness is very low for every zombie (see: mostly every other zombie game)

How do they bite through it? It would be great to have damaged masks (broken faceplate with zombie inside) etc but we don't right now.

> Zombies historically don't take damage from pressure in SS13. There are usually ships that are spaced with active zombies on them.

I'm just trying to counter the zombies spacing the station intentionally.

> Stunning on conversion is a horrible idea. Especially one for that long.

I reduced the stun a lot

keronshb commented:
> How do they bite through it? It would be great to have damaged masks (broken faceplate with zombie inside) etc but we don't right now.

SS13 has zombies scratch instead. It's really more of a flavor thing.
There really is no full reason to remove their helmet & mask on zombification. Zombies with hats do exist in media.

> I'm just trying to counter the zombies spacing the station intentionally.

Please leave them as is, thank you.

> I reduced the stun a lot

I still don't feel like the stun is the way to go. A more robust way to handle it would probably leave them dead for a few seconds instead, then revive as a zombie.

tom-leys commented:
Zombification now occurs after the victim dies, not just crits. You can hurry the process by biting them some more. Also the growing pain of the zombie virus might leave your victims alive for a few minutes as the DoT mounts.

https://user-images.githubusercontent.com/5285589/236717853-dd13dd03-2a76-44bd-8e67-4c55fd7081dd.mp4

keronshb commented on Content.Server/Zombies/ZombifyOnDeathSystem.cs:
```diff
@@ -101,7 +107,7 @@ public void ZombifyEntity(EntityUid target, MobStateComponent? mobState = null)
             //get diseases, breath, be thirst, be hungry, or die in space
             RemComp<RespiratorComponent>(target);
-            RemComp<BarotraumaComponent>(target);
+            // RemComp<BarotraumaComponent>(target); Actually yes, they should die in space (to prevent zombie spacing)
```
```suggestion
             RemComp<BarotraumaComponent>(target);
```
Please uncomment this.

tom-leys replied:
The vacuum DoT is mostly balanced by the zombie heal over time. Are you sure?

tom-leys replied:
Done.

keronshb commented on Content.Server/Zombies/ZombifyOnDeathSystem.cs:
```diff
@@ -169,6 +175,9 @@ public void ZombifyEntity(EntityUid target, MobStateComponent? mobState = null)
             _serverInventory.TryUnequip(target, "gloves", true, true);
             //Should prevent instances of zombies using comms for information they shouldnt be able to have.
             _serverInventory.TryUnequip(target, "ears", true, true);
+            //Shouldn't be wearing a helmet. How are you going to bite things?
+            _serverInventory.TryUnequip(target, "head", true, true);
+            _serverInventory.TryUnequip(target, "mask", true, true);
```
```suggestion
```

Please remove these

tom-leys replied:
Fine. Done.

Myakot commented:
Why does the new code make zombies regenerate cold damage if they are literal undead and their damagemodifier set cold damage to 0, making them unable to actually receive cold damage?

tom-leys commented:
> Why does the new code make zombies regenerate cold damage if they are literal undead and their damagemodifier set cold damage to 0, making them unable to actually receive cold damage?

I'd never actually looked that up. Turns out they re very resistant but not immune to cold.

```
- type: damageModifierSet
  id: Zombie #Blunt resistent and immune to biological threats, but can be hacked apart and burned
  coefficients:
    Blunt: 0.5
    Slash: 0.7
    Piercing: 0.7
    Shock: 1.5
    Cold: 0.2
    Heat: 2.0
    Poison: 0.0
    Radiation: 0.0
```

deltanedas commented:
zombies spacing everywhere is more of an administrative issue imo since they are meant to infect people not just do as much damage as possible

tom-leys commented:
> zombies spacing everywhere is more of an administrative issue imo since they are meant to infect people not just do as much damage as possible

Well, it would be nice if the game mechanics support the play we want. But the spacing counter got removed anyway.

OctoRocket commented:
I don't think that zombies should really be buffed, if they are they should change how strong they are based on player count or something so that lowpop mrp doesn't get steamrolled.

EmoGarbage404 commented:
The healing over time changes and conversion on death change is good, but I'm not really sure about the infection rate change. Ideally, it shouldn't feel like every hit is a guaranteed infection, which is how a lot of people feel right now.

tom-leys commented:
> I don't think that zombies should really be buffed, if they are they should change how strong they are based on player count or something so that lowpop mrp doesn't get steamrolled.

Yeah, nether do I. This PR goes like this:

I wanted to nerf zombies in space, to discourage them spacing the station:
- So I added slow vacuum damage

Also wanted to make them more identifiable and make it more rational that they were biting stuff:
- So I removed their helmet and mask

And I wanted to make the transition to zombies less sudden:
- So I added a stun on zombification.

Then I was asked to add buffs to make the nerfs less painful early on
- More virulent virus
- Stun is instead a go into crit
- Zombie regen

Finally it was demanded that I remove the nerfs
- Helmet and mask back on
- Space damage removed

As for the MRP servers, the one I know of (parkstation) has AI roles which are extremely OP vs zombies as they can scout and bolt doors in advance of the horde.

tom-leys commented:
> The healing over time changes and conversion on death change is good, but I'm not really sure about the infection rate change. Ideally, it shouldn't feel like every hit is a guaranteed infection, which is how a lot of people feel right now.

Done

keronshb commented on Content.Server/Zombies/ZombifyOnDeathSystem.cs:
```diff
@@ -221,6 +234,16 @@ public void ZombifyEntity(EntityUid target, MobStateComponent? mobState = null)
+            // No longer waiting to become a zombie:
+            RemComp<PendingZombieComponent>(target);
+
+            // Zombiefication takes a second. Groan on the floor for a bit.
+            if (EntityManager.TryGetComponent<StatusEffectsComponent>(target, out var status))
+            {
+                _stunSystem.TryStun(target, TimeSpan.FromSeconds(1.0), true, status);
+                _stunSystem.TryKnockdown(target, TimeSpan.FromSeconds(0.8), true,
+                    status);
+            }
```
```suggestion
```

This should be removed or changed.

 As it is right now you die > revive as a zombie > get up for 1/4 of a second > get stunned for 1 second > get up. It feels very awkward, especially if a zombie is clawing you on death and you turn into a zombie on the next hit, which registers as someone falling down twice in a row. 

They're already dead before turning into a zombie so there's no need for a stun.

Alternatively, it should be a number of seconds _after_ death before they turn if you wanted to time gate the transformation.

keronshb commented on Content.Server/Zombies/ZombifyOnDeathSystem.cs:
```diff
@@ -54,6 +58,9 @@ public sealed class ZombifyOnDeathSystem : EntitySystem
         [Dependency] private readonly SharedCombatModeSystem _combat = default!;
         [Dependency] private readonly IChatManager _chatMan = default!;
         [Dependency] private readonly IPrototypeManager _proto = default!;
+        [Dependency] private readonly StunSystem _stunSystem = default!;
```
```suggestion
```

Remove this also with the stun below.

keronshb reviewed (changes requested):
If you want to make the zombies a bit less stealthy, you can have them emote/force say randomly on top of the random sound they already play. That way there's not only a sound cue, but a visual one with text.

Otherwise, just the changes then this is good to go.

tom-leys commented:
> If you want to make the zombies a bit less stealthy, you can have them emote/force say randomly on top of the random sound they already play. That way there's not only a sound cue, but a visual one with text.
> 
> Otherwise, just the changes then this is good to go unless something else is needed.

That would be cool, but a little bit beyond me right now. Everything else is done.
